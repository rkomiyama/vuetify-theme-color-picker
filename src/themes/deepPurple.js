export default {
  name: "Deep Purple",
  color: "#673ab7",
  themes: {
    light: {
      primary: "#673ab7",
      secondary: "#7C4DFF",
      accent: "#6200EA",
      error: "#FF5252",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FFC107"
    },
    dark: {
      primary: "#673ab7",
      secondary: "#7C4DFF",
      accent: "#AA00FF",
      error: "#6200EA",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FB8C00"
    }
  }
};
