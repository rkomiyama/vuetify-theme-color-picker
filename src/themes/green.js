export default {
  name: "Green",
  light: true,
  dark: false,
  color: "#4caf50",
  themes: {
    light: {
      primary: "#4caf50",
      secondary: "#8bc34a",
      accent: "#03a9f4",
      error: "#f44336",
      warning: "#ffc107",
      info: "#00bcd4",
      success: "#009688"
    },
    dark: {
      primary: "#4caf50",
      secondary: "#8bc34a",
      accent: "#03a9f4",
      error: "#f44336",
      warning: "#ffc107",
      info: "#00bcd4",
      success: "#009688"
    }
  }
}
