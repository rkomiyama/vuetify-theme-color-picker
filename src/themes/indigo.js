export default {
  name: "Indigo",
  color: "#3F51B5",
  themes: {
    light: {
      primary: "#3F51B5",
      secondary: "#9FA8DA",
      accent: "#304FFE",
      error: "#FF5252",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FFC107"
    },
    dark: {
      primary: "#3F51B5",
      secondary: "#9FA8DA",
      accent: "#304FFE",
      error: "#FF5252",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FB8C00"
    }
  }
};
