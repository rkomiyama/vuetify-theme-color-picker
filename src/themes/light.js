export default {
  name: "Light",
  light: true,
  dark: false,
  color: "#000000",
  themes: {
    light: {
      primary: '#1976D2',
      secondary: '#424242',
      accent: '#82B1FF',
      error: '#FF5252',
      info: '#2196F3',
      success: '#4CAF50',
      warning: '#FFC107',
    },
    dark: {
      primary: '#2196F3',
      secondary: '#424242',
      accent: '#FF4081',
      error: '#FF5252',
      info: '#2196F3',
      success: '#4CAF50',
      warning: '#FB8C00',
    }
  }
}