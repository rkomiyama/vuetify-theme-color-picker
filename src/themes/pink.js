export default {
  name: "Pink",
  color: "#E91E63",
  themes: {
    light: {
      primary: "#E91E63",
      secondary: "#F8BBD0",
      accent: "#FF4081",
      error: "#FF5252",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FFC107"
    },
    dark: {
      primary: "#E91E63",
      secondary: "#F8BBD0",
      accent: "#FF4081",
      error: "#FF5252",
      info: "#2196F3",
      success: "#4CAF50",
      warning: "#FB8C00"
    }
  }
};
