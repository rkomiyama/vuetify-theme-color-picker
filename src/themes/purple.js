export default {
  name: "Purple",
  light: true,
  dark: false,
  color: "#673ab7",
  themes: {
    light: {
      primary: "#673ab7",
      secondary: "#9c27b0",
      accent: "#e91e63",
      error: "#f44336",
      warning: "#ffc107",
      info: "#00bcd4",
      success: "#4caf50"
    },
    dark: {
      primary: "#673ab7",
      secondary: "#9c27b0",
      accent: "#e91e63",
      error: "#f44336",
      warning: "#ffc107",
      info: "#00bcd4",
      success: "#4caf50"
    }
  }
}
