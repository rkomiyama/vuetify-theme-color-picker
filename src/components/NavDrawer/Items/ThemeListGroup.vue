<template>
  <v-list-group
    prepend-icon="mdi-palette-advanced"
    ripple
  >
    <template v-slot:activator>
      <v-list-item-content>
        <v-list-item-title>
          Themes
        </v-list-item-title>
      </v-list-item-content>
    </template>
    <v-list-item-group
      v-model="themeOption"
      mandatory
    >
      <v-list-item
        v-for="(theme, i) in themes"
        class="pl-10"
        @click="clickListItem(theme)"
        link
        :key="i"
      >
        <v-list-item-action>
          <v-icon :color="theme.color">
            {{ `mdi-circle${theme.name === "Light" ? "-outline" : ""}` }}
          </v-icon>
        </v-list-item-action>
        <v-list-item-content>
          <v-flex class="d-flex">
            <v-list-item-title>{{ theme.name }}</v-list-item-title>
            <v-switch
              v-if="theme.name === 'Light'"
              v-model.lazy="darkOrLight"
              @change="clickSwitch('Light')"
              value="Light"
              @click.stop="clickListItem(theme)"
              inset
            ></v-switch>
            <v-switch
              v-else-if="theme.name === 'Dark'"
              v-model.lazy="darkOrLight"
              @change="clickSwitch('Dark')"
              value="Dark"
              @click.stop="clickListItem(theme)"
              inset
            ></v-switch>
          </v-flex>
        </v-list-item-content>
      </v-list-item>
    </v-list-item-group>
  </v-list-group>
</template>

<script>
import themes from '@/themes'

export default {
  name: "ThemeListGroup",
  data () {
    return {
      themeOption: 0,
      darkOrLight: ['Light'],
      darkTheme: false,
      themesSwapped: false,
      themes
    }
  },
  methods: {
    swapColors (newTheme) {
      this.$vuetify.theme.themes.light = newTheme.themes.light
      this.$vuetify.theme.themes.dark = newTheme.themes.dark
    },
    clickSwitch (themeName) {
      const otherThemeName = themeName === "Dark" ? "Light" : "Dark"
      this.darkOrLight = this.darkOrLight[0] === themeName ? [otherThemeName] : [themeName]
      this.$vuetify.theme.light = this.darkOrLight[0] === "Light"
      this.$vuetify.theme.dark = this.darkOrLight[0] === "Dark"
    },
    clickListItem (theme) {
      if (theme.name === "Light" || theme.name === "Dark") {
        this.clickSwitch(theme.name)
      }
      this.swapColors(theme)
    }
  }
}
</script>

