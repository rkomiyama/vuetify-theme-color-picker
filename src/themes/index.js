import light from './light'
import dark from './dark'
import red from './red'
import purple from './purple'
import green from './green'

export default {
  light,
  dark,
  red,
  purple,
  green
}
