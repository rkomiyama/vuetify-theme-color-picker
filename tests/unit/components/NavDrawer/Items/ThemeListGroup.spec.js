import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer';
import themes from '@/themes'
import ThemeListGroup from '@/components/NavDrawer/Items/ThemeListGroup.vue'

Vue.use(Vuetify)
let wrapper

describe('ThemeListGroup.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(ThemeListGroup)
    wrapper.vm.$vuetify = {
      theme: {
        themes: {
          light: themes.light,
          dark: themes.dark
        }
      }
    }
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  it('calls swapColors', () => {
    wrapper.vm.swapColors(themes.red)
    expect(wrapper.vm.$vuetify.theme.themes.light).toEqual(themes.red.themes.light)
  })

  it('calls clickListItem', () => {
    wrapper.vm.clickListItem(themes.dark)
    expect(wrapper.vm.$data.darkOrLight).toEqual(['Dark'])
    wrapper.vm.clickListItem(themes.red)
    expect(wrapper.vm.$vuetify.theme.themes.dark).toEqual(themes.red.themes.dark)
    wrapper.vm.clickListItem(themes.dark)
    expect(wrapper.vm.$data.darkOrLight).toEqual(['Light'])
    wrapper.vm.clickListItem(themes.light)
    expect(wrapper.vm.$data.darkOrLight).toEqual(['Dark'])
  })

  it('ThemeListGroup renders the same HTML', () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
    if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
})
