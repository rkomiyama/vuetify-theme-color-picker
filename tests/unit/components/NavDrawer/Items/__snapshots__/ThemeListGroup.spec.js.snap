// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeListGroup.vue ThemeListGroup renders the same HTML 1`] = `
<v-list-group-stub color="primary" activeClass="" appendIcon="$vuetify.icons.expand" prependIcon="mdi-palette-advanced" ripple="true">
  <v-list-item-group-stub value="0" activeClass="v-item--active" mandatory="true">
    <v-list-item-stub activeClass="" link="true" tag="div" class="pl-10">
      <v-list-item-action-stub>
        <v-icon-stub color="#000000">
          mdi-circle-outline
        </v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-flex-stub tag="div" class="d-flex">
          <v-list-item-title-stub>Light</v-list-item-title-stub>
          <v-switch-stub errorCount="1" errorMessages="" messages="" rules="" successMessages="" value="Light" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" inset="true"></v-switch-stub>
        </v-flex-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub activeClass="" link="true" tag="div" class="pl-10">
      <v-list-item-action-stub>
        <v-icon-stub color="#000000">
          mdi-circle
        </v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-flex-stub tag="div" class="d-flex">
          <v-list-item-title-stub>Dark</v-list-item-title-stub>
          <v-switch-stub errorCount="1" errorMessages="" messages="" rules="" successMessages="" value="Light" backgroundColor="" ripple="true" valueComparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" inset="true"></v-switch-stub>
        </v-flex-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub activeClass="" link="true" tag="div" class="pl-10">
      <v-list-item-action-stub>
        <v-icon-stub color="#f44336">
          mdi-circle
        </v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-flex-stub tag="div" class="d-flex">
          <v-list-item-title-stub>Red</v-list-item-title-stub>
          <!---->
        </v-flex-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub activeClass="" link="true" tag="div" class="pl-10">
      <v-list-item-action-stub>
        <v-icon-stub color="#673ab7">
          mdi-circle
        </v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-flex-stub tag="div" class="d-flex">
          <v-list-item-title-stub>Purple</v-list-item-title-stub>
          <!---->
        </v-flex-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
    <v-list-item-stub activeClass="" link="true" tag="div" class="pl-10">
      <v-list-item-action-stub>
        <v-icon-stub color="#4caf50">
          mdi-circle
        </v-icon-stub>
      </v-list-item-action-stub>
      <v-list-item-content-stub>
        <v-flex-stub tag="div" class="d-flex">
          <v-list-item-title-stub>Green</v-list-item-title-stub>
          <!---->
        </v-flex-stub>
      </v-list-item-content-stub>
    </v-list-item-stub>
  </v-list-item-group-stub>
</v-list-group-stub>
`;
